Sources/ForageSDK/Foundation/Network/Provider.swift

```diff
@@ -91,31 +87,22 @@ class Provider {
     }
 
     private func middleware<T: Decodable>(model: T.Type, data: Data?, response: URLResponse?, error: Error?, completion: @escaping (Result<T, Error>) -> Void) {
-        var httpResponse: HTTPURLResponse?
-
-        processResponse(response: response) { result in
-            switch result {
-            case let .success(response):
-                self.logger?.info(
-                    "Received \(response.statusCode) response from \(self.getResponseUrlPath(response))",
-                    attributes: ["endpoint": httpResponse?.url?.path]
-                )
-                httpResponse = response
-            case let .failure(error):
-                self.logger?.error("Failed to process response", error: error, attributes: nil)
-                return completion(.failure(error))
-            }
+        if let error = error {
+            let httpResponse = response as? HTTPURLResponse
+            let wrappedError = NSError(domain: "Error: \(error)", code: httpResponse?.statusCode ?? 500, userInfo: nil)
+            self.logger?.error("Failed to process error for \(self.getResponseUrlPath(httpResponse))", error: wrappedError, attributes: nil)
+            return completion(.failure(wrappedError))
```

Comment:
why `completion(.failure(wrappedError))` instead of the original `completion(.failure(error))`?